src/test/groovy/net/fabricmc/loom/test/integration/LegacyProjectTest.groovy

```diff
@@ -86,8 +86,41 @@ class LegacyProjectTest extends Specification implements GradleProjectTestTrait
 		'1.3.2'			| _
 	}
 
+	// TODO: need intermediaries for this test to succeed
 	@Unroll
-	def "Ancient minecraft (minecraft #version)"() {
+	def "Ancient minecraft merged (minecraft #version)"() {
+		setup:
+		def gradle = gradleProject(project: "minimalBase", version: PRE_RELEASE_GRADLE)
+		gradle.buildGradle << """
+				loom {
+                    noIntermediateMappings()
+                }
+
+                dependencies {
+                    minecraft "com.mojang:minecraft:${version}"
+                    mappings loom.layered() {
+						// No names
+					}
+
+                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                }
+			"""
+
+		when:
+		def result = gradle.run(task: "configureClientLaunch")
+
+		then:
+		result.task(":configureClientLaunch").outcome == SUCCESS
```

Comment:
We really need to test something more substanial here, a `build` at least with a small mod that compiles against both common and client classes.. It will likely need its own test project + file. Im not against adding some open source mappings from another project into looms repo. 

The tests above are mostly to ensure that the client assets are downloaded correctly, and do not actually test the jars created.

Reply:
I'll be honest, I have no clue how to write proper tests for this. I saw the legacy projects tests and just copied the ancient minecraft one and changed it up a little. Perhaps it's best I revert my changes to this file and ask you or someone else who knows what they're doing to write the tests.

I can provide some mappings for those though, Ornithe's Stitch fork should have everything in place now to generate those.

Reply:
Im quite happy to either help you write them, or write it myself. The tests arent that scary, its just like running a normal gradle project, you just need to set it up, and then validate that the output is as you expect.

For this I think it would be nice to use some real mappings, (they could just be included in the loom repo as a zip file). And then have a very simple mod that just compiles against some clases.

Reply:
A modified version of babric intermediary could be used to test this.